pkg/projectfile/nameval.go

```diff
@@ -0,0 +1,63 @@
+package projectfile
+
+import (
+	"bytes"
+
+	"gopkg.in/yaml.v2"
+)
+
+type NameVal struct {
+	Name  string `yaml:"name"`
+	Value string `yaml:"value"`
+}
+
+func (nv *NameVal) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	data := make(map[string]interface{})
+	if err := unmarshal(&data); err != nil {
+		return err
+	}
+
+	if len(data) == 1 {
+		ssData, err := sanitizeMap(data)
+		if err != nil {
+			return err
+		}
+
+		for k, v := range ssData {
+			nv.Name, nv.Value = k, v
+			break
+		}
```

Comment:
I think we can address the caveats by simply doing a regular unmarshal when we encounter a "name" field.

We could represent the default struct without a `UnmarshalYAML()` method, and then use composition to create another struct that has `UnmarshalYAML()`, when this one encounters "name" we just defer to the default marshal behaviour on the embedded struct.

Hit me up on Slack so we can zoom if I'm not being clear. Key point is I think we can still achieve the shortcut mechanic you want without sacrificing error reporting.

I do agree manually unmarshalling the actual data is something we want to avoid. We are effectively piggybacking off UnmarshalYAML to basically direct the unmarshalling mechanic one of two ways.